README.md

```diff
@@ -10,10 +21,73 @@ Error code  | Error Name | Description
  1   | Server Exception | Unknown error from the server 
  2   | Bad request | The request make do not follow expected payload 
 
-# Messages
+# Handshake
+When a client connect to a Lama node. It need to init a handshake, the sequence of messages does not evolve.
+
+## Sequence
+Client => Server : Authentification
+Server => Client : Supported protocol version
+Client => Server : Choose a protocol version to use
+
+## Authentification
+
+Client call the server, for authentification.
+```
+auth_method => UINT8
+identity_name => STRING
+identity_key => [BUFFER]
+```
+
+Field | Description
+------|--------------
+auth_method | Version of the auth method (always 0 currently).
+identity_name | The name of the client
+identity_key | Optional buffer containing a key for identifily uniquely a client
+
+## Supported protocol version
+Server answer (if success) with the protocol version supported
+
+```
+max_version_major_supported=UINT8
+max_version_minor_supported=UNIT8
+min_version_major_supported=UINT8
+min_version_minor_supported=UINT8
+```
```

Comment:
Not sure about using "patch" level, I don't really know what kind of meaning "patch" on a RFC of protocol :thinking:

Reply:
In theory, patch does not "add" features, so in theory a server handle 1.3.0 could handle 1.3.1, but like I said before, I do not see what patch mean for protocol version

Reply:
I see, maybe, patch could handle "protocol change", but not the spec, (like a text mistake, or a typo, you change the original protocol definition, but the protocol does not evolve).